Remotion/Data/DomainObjects/Queries/Configuration/QueryDefinition.cs

```diff
@@ -86,8 +91,8 @@ public class QueryDefinition
   ///   <paramref name="statement"/> is an empty string.
   /// </exception>
   /// <exception cref="System.ArgumentOutOfRangeException"><paramref name="queryType"/> is not a valid enum value.</exception>
-  public QueryDefinition (string queryID, StorageProviderDefinition storageProviderDefinition, string statement, QueryType queryType)
-    : this(queryID, storageProviderDefinition, statement, queryType, null)
+  public QueryDefinition (string queryID, StorageProviderDefinition storageProviderDefinition, string statement, QueryType queryType, IReadOnlyDictionary<string, object>? metaData = null)
+    : this(queryID, storageProviderDefinition, statement, queryType, null, metaData)
```

Comment:
I think we should merge the two constructors, because `collectionType` is also effectively an optional parameter and should be implemented as such. (The current implementation is probably older than the "optional parameter" syntactic feature.)
It just seems awkward to mix two styles of optional parameters.